src/System.Private.ServiceModel/tests/Common/Scenarios/ScenarioTestHelpers.cs

```diff
@@ -153,6 +153,42 @@ public static ComplexCompositeType GetInitializedComplexCompositeType()
 
         return compositeObject;
     }
+
+    /// <summary>
+    /// Closes com objects in the order passed in if not already closed.
+    /// If Close fails for Timeout or Communication exception then Aborts.
+    /// </summary>
+    /// <param name="objects">Any communication objects that need to be cleaned up.
+    /// In the order in which they need to be cleaned up.</param>
+    public static void CloseCommunicationObjects(params ICommunicationObject[] objects)
+    {
+        foreach(ICommunicationObject comObj in objects)
+        {
+            try
+            {
+                if(comObj != null)
+                {
+                    continue;
+                }
+                if(comObj.State == CommunicationState.Opened)
+                {
+                    comObj.Close();
+                }
+                else if (comObj.State != CommunicationState.Closed)
```

Comment:
Should this be an `if` rather than `else if`? After the call to Close(), we're expecting the channel to be Closed regardless of whether or not the channel was in an Opened state when we entered the method.